File: app/src/main/java/com/hjc/base/bean/ConcertBean.java
package com.hjc.base.bean;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import java.util.Objects;

public class ConcertBean extends BaseObservable {
    private String title;
    private String author;

    @Bindable
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Bindable
    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (this.getClass() == obj.getClass()) {
            return true;
        }
        ConcertBean concert = (ConcertBean) obj;
        return Objects.equals(title, concert.title) && Objects.equals(author, concert.author);
    }

    @Override
    public int hashCode() {
        return Objects.hash(title, author);
    }
}

File: module-home/src/main/java/com/hjc/module_home/http/entity/ConcertBean.kt
package com.hjc.module_home.http.entity

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import java.util.*

class ConcertBean : BaseObservable() {
    @get:Bindable
    var title: String? = null

    @get:Bindable
    var author: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null) {
            return false
        }
        if (this.javaClass == other.javaClass) {
            return true
        }
        val concert = other as ConcertBean
        return title == concert.title && author == concert.author
    }

    override fun hashCode(): Int {
        return Objects.hash(title, author)
    }
}